service/src/main/java/uk/gov/hmcts/reform/fpl/service/OthersService.java

```diff
@@ -65,8 +70,13 @@ public Others prepareOthers(CaseData caseData) {
         caseData.getAllOthers().forEach(element -> {
             if (element.getValue().containsConfidentialDetails()) {
                 Other confidentialOther = getConfidentialItemToAdd(caseData.getConfidentialOthers(), element);
-
-                others.add(element(element.getId(), addConfidentialDetails(confidentialOther, element)));
+                Element<Other> otherElementWithConfidentialDetails = element(element.getId(),
+                    addConfidentialDetails(confidentialOther, element));
+                // copying the representedBy to confidential others since addConfidentialDetails will remove them
```

Comment:
probably good if we can raise a technical ticket to check this?

Reply:
Is it maybe because of 

private final List<Element<UUID>> representedBy = new ArrayList<>();

in Other.java?